src/main/java/org/commonjava/util/sidecar/services/ArchiveRetrieveService.java

```diff
@@ -0,0 +1,279 @@
+                                                         {
+                                                             return false;
+                                                         }
+                                                         if ( exception instanceof NoHttpResponseException )
+                                                         {
+                                                             logger.info( "NoHttpResponse start to retry times:"
+                                                                                          + executionCount );
+                                                             return true;
+                                                         }
+                                                         if ( exception instanceof SocketTimeoutException )
+                                                         {
+                                                             logger.info( "SocketTimeout start to retry times:"
+                                                                                          + executionCount );
+                                                             return true;
+                                                         }
+                                                         if ( exception instanceof ConnectionPoolTimeoutException )
+                                                         {
+                                                             logger.info( "ConnectionPoolTimeout start to retry times:"
+                                                                                          + executionCount );
+                                                             return true;
+                                                         }
+                                                         return false;
+                                                     } );
+        client = builder.build();
+    }
+
+    @PreDestroy
+    public void destroy()
+    {
+        try
+        {
+            File downloadDir = new File( localRepository.get() );
+            List<File> downloads = Files.walk( downloadDir.toPath() )
+                                        .filter( Files::isRegularFile )
+                                        .map( Path::toFile )
+                                        .collect( Collectors.toList() );
+            for ( File content : downloads )
+            {
+                content.delete();
+            }
+            downloadDir.delete();
+        }
+        catch ( IOException e )
+        {
+            logger.error( "Failed to clean up downloads", e );
+        }
+        finally
+        {
+            IOUtils.closeQuietly( client, null );
+        }
+    }
+
+    public boolean decompressArchive()
+    {
+        if ( archiveApi.isEmpty() )
+        {
+            return false;
+        }
+        String buildConfigId = System.getProperty( BUILD_CONFIG_ID );
```

Comment:
maybe you can extract this System.getProperty(BUILD_CFG_ID) to a single method as it is used several times. So if we will use some alternative way we can easily change it universally.

Reply:
@ligangty to check